Wrappers/Python/cil/optimisation/algorithms/SIRT.py

```diff
@@ -133,11 +134,18 @@ def fix_weights(self):
         
         # fix for possible +inf, -inf, nan values 
         for arr in [self.M, self.D]:  
- 
-            tmp = arr.as_array()
-            arr_replace = numpy.isfinite(tmp)
-            tmp[~arr_replace] = 1.0     
-            arr.fill(tmp)                     
+            self._remove_nan_or_inf(arr, replace_with=1.0)
+
+    def _remove_nan_or_inf(self, datacontainer, replace_with=1.0):
+        """Replace nan and inf with a given value."""
```

Comment:
```suggestion
        """Replace nan and inf in datacontainer with a given value.
        
        Parameters:
        -------------
        
        datacontainer: DataContainer, BlockDataContainer
        
        replace_with: float, default 1.0
            Value to replace elements that evaluate to NaN or inf
            
            
        In case the input datacontainer is a :code:`BlockDataContainer` the substitution is executed for each container in the :code:`BlockDataContainer`."""
```